enhancements/network-admission/allow-external-ip-overrides.md

```diff
@@ -0,0 +1,141 @@
+An admin user can do so by configuring the service specifications' [externalIPs](https://kubernetes.io/docs/concepts/services-networking/service/#external-ips "externalips") field to a value that falls outside of the allowed range.
+
+## Motivation
+
+In 4.1, the **externalIPNetworkCIDR** config flag is missing from the network configuration. 
+This combined with the fact that the external IP range check admission plugin is not registered in 4.1, creates a security 
+issue that all external IPs are allowed. There is a parallel proposal to block all external IPs to plug this security hole 
+in a 4.1 z-stream release. But when we block all external IPs, we also need to provide admin users the ability to specify 
+the external IPs where safe.
+
+In 4.2, the **externalIPNetworkCIDR** will be re-added as a [configuration parameter](https://github.com/openshift/api/blob/master/config/v1/types_network.go#L54) in the network-config yaml for the openshift-sdn. 
+However, users who would upgrade from 4.1 z-stream where admin over-ride of external IPs is not just supported but needed 
+to 4.2, would expect their service configurations to work as is. This proposal therefore is to allow the admin over-ride 
+in 4.x as a whole.
+
+### Goals
+
+- Provide a mechanism for cluster admin user(s) to over-ride cluster configured values of ranges of external IPs for a service. 
+<a name="admin-user">An admin user here is defined as someone who is given a special permission - see details below</a>.
+
+- As a corollary, non-admin user should not be able to patch/update the service spec with external IPs that fall out of the 
+range specified at the cluster level.
+
+- Both admin and non-admin users should still be able to specify external IPs for a service that fall within the range 
+specified at the cluster level.
+
+### Non-Goals
+
+- Escalation of privileges by causing service traffic to be routed to a malicious external IP address.
+
+## Proposal
+
+- As an administrator with [sufficient privilege](#admin-user) on a 4.x Openshift cluster, I want the ability to 
+specify external IPs which may fall out of the range specified by cluster administrators for services belonging to my app.
+
+This proposal is to add an RBAC check in case the external IPs specified for the service don't fall in the range of the 
+cluster specified **externalIPNetworkCIDR**. This is needed particularly in 4.1, where the external IPs will be blocked 
+completely. In case, there are validation errors as a result of the specified external IP(s) not being in the valid range, 
+the RBAC checker will validate that the user has permissions to create a dummy subresource for the service in question. 
+If the result of that check is an Allow, the user is allowed to set the external IP(s) on the service.
+
+Code snippet that performs the actual RBAC check:
+
+	``` go
+    func (r *externalIPRanger) checkAccess(attr admission.Attributes) (bool, error) {
```

Comment:
This looks right to me. Please defend down below that this is cluster-scoped because setting in one namespace had an impact across other namespaces. That will be enough detail to jog our memories.

@enj you own a pull making this permission escalating for scopes.